### DISCUSSION TITLE
Adding Pentair 011013 IntelliFlo VS 3050 pump - fails

### DISCUSSION THREAD

**Discussion by AnomanderRake69** [rstrouse/nodejs-poolController-dashPanel#66] [Q&A] [answered]
Hi there, just getting started with your pool controller, and I'm stuck on adding a pump. Mine is a Pentair 011013 IntelliFlo VS 3050, and upon clicking "Save" in the dash panel, I get "Message aborted after 3 attempt(s)". I can't seem to delete the partially added entry either, receiving the same message. I've poked around on the project page and other sites for a few days and I have no idea where to go next.

I've attached a replay which I think captures the event.

Thanks for any guidance.
[replay - add pump.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10097798/replay.-.add.pump.zip)

**Comment by tagyoureit** [comment, marked as answer]
Hi, I do see the EasyTouch and pump communicating.  I also see an 'Error: undefined' in the logs but the packet capture doesn't have all of the communications.

Can you please do the following:
1. Start a replay capture
2. Stop the njsPC app
3. delete the `./data/pool*.json` files
4. restart the app
5. let the entire configuration finish loading
6. if the pump is still not present, you can try to add it like you did previously; but hoping it gets added here
7. stop the capture and attach the new replay

**Reply by AnomanderRake69** [reply]
Thanks @tagyoureit. Here's a more complete capture that includes the complete configuration. The pump was not detected by the startup configuration detection, so I attempted to add it manually, and received the same response after hitting "Save". I have tried both address 1 and 2 for the pump, same result. 

Looking at the packet log in message manager (which is awesome, btw), I see message #311 is highlighted in orange as invalid, and is tagged to the pump. Unfortunately, I'm very new to your app and still navigating my way around, so I don't have much more insight than that.

In case its not said enough, thank you for doing this. I bought my house a few months ago, acquired a pool with it for the first time in my life, and was instantly depressed by the quality of the Pentair Easy Touch, especially the interface. If I can get this working, it will tide me over until I can build myself a custom Nixie setup :)

[replay full.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10098616/replay.full.zip)

**Reply by tagyoureit** [reply]
Hi, it's a bit strange that the OCP (outdoor control panel = EasyTouch) isn't responding to the pump config requests... neither when you request the config or try to set it.

Please try these two basic things and see if it makes a difference (no need to do a replay capture)...
1. Please check the pump settings directly on the EasyTouch and see what shows up when you look through the menus.
2. Also try a hard reset of the OCP.  Power cycle it/use the reset button and see if you get different behavior.  

Start with those and let me know if it wakes the controller up to respond to the pump settings.

**Reply by AnomanderRake69** [reply]
1. Not sure what to look for in the menus - the only thing I could find is the pump speed settings, so I've attached 4 shots of what those look like.
2. I did both a reset from the diagnostics menu as well as by flipping the circuit breakers, but no change in behavior.

I've noticed that my IC20 SWCG is not automatically identified either. Would this kind of issue be indicative of a bad RS485 adapter? I just bought mine, its the Elfin EW11. There was a long lead time on this one as it shipped from China, I'd hate to have to order a different one, but if I have to then I have to.

Thanks again.
![pump1](https://user-images.githubusercontent.com/10177469/204408475-8d503e6d-5e30-4972-a059-5e166ef201a5.jpg)
![pump2](https://user-images.githubusercontent.com/10177469/204408483-836bed19-f393-4ac2-9edf-bfa30b988210.jpg)
![pump3](https://user-images.githubusercontent.com/10177469/204408489-99b50028-39e9-4fe1-a4ee-29248d7d244a.jpg)
![pump4](https://user-images.githubusercontent.com/10177469/204408492-9ed3965b-e2d1-4a32-9123-47d600586526.jpg)

**Reply by rstrouse** [reply]
It is very odd that the panel is not responding to action 216 requests for either potential pump.  So can I get you to perform the following steps.

1. Start a replay capture by clicking on the hamburger menu on the upper left.  Select the logging tab and press the Capture Replay button.  When the dialog comes up ensure the Capture Configuration Reload is unchecked.
2. Go out to the pad and assign the 3/4 page to SOLAR as the circuit and 2570 as the RPM.
3. Assign 4/4 page to NONE as the circuit.
4. Return to the logging tab in dashPanel and press the End Capture button
5. Upload the replay zip file that was downloaded to your browser here.

Lets see if this firmware is calling a different message to set and get the configuration for the pump.

**Reply by AnomanderRake69** [reply]
Here you go, followed the 5 steps above and changes the 3/4 and 4/4 settings for my pump. I loaded this in the message manager, but I'm not quite good enough to decipher it yet. Hopefully you can get something out of it.

And your suggestion of different message numbers for different firmware versions would not surprise me at all. However, both the OCP and the pump have to be aware of what the message number taxonomy looks like, in order to communicate? Maybe there is a range of valid pump status requests to account for changing firmwares over time.
[replay - pump changes.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10118569/replay.-.pump.changes.zip)

**Reply by rstrouse** [reply]
I did not see any configuration messages in that output for anything.  So lets verify a couple of things here.

First the screen shots you show above do not match anything that I can find in the manual.  The screenshots look like a configuration for a SuperFlo VS pump that EasyTouch does not really support.  Is this an EasyTouch 2 8 for certain?

Can I get you to verify the settings from page 33 of this manual.  There should be 8 pages of speed settings for the pump.
https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/automation/easytouch/manual/easytouch-control-system-8-and-4-users-guide.pdf

**Reply by AnomanderRake69** [reply]
Well I feel stupid and embarrassed. As I said, I'm new to pools and new to Pentair, but I should have paid closer attention to this. njsPC identifies my OCP as an "EasyTouch 2 8" as you indicate above, but I kind of knew that is not what I have. The SKU on the inside of the door is 520592 which on Pentair's product page [here](https://www.pentair.com/en-us/products/residential/pool-spa-equipment/pool-automation/easytouch_controlsystems/sku/520592.html) is an "EASYTOUCH SYSTEMS WITH INTELLICHLOR TRANSFORMER AND INTELLICHLOR CELL,  EasyTouch 4PSC-IC20 - Single Body".

![sku label](https://user-images.githubusercontent.com/10177469/204703813-2d8265ca-343c-41d8-a4c7-f8bdaa704b1a.jpg)

The test menu identifies this as as an EasyTouch 4P:

![easytouch 4p](https://user-images.githubusercontent.com/10177469/204703936-2aa8f593-fac0-4653-8fd7-35341eee46bd.jpg)

And the high voltage compartment only has 4 relays mounted:

![high volt](https://user-images.githubusercontent.com/10177469/204704032-4181eae6-a7e4-410b-ac09-90d532a3d9ea.jpg)

So I guess the question is if I have a strange/obscure Pentair SKU that maybe you guys have not seen before? Is there anyway for me to override the type of OCP that the software automatically detects?

**Reply by rstrouse** [reply]
Pull njsPC it should now recognize the controller as an EasyTouch 4P.  We have not seen one of these as they are very old.  Do you have a controller that is not the QuickTouch 4.  This controller used to come with a remote that has a little LCD like the outdoor control panel?

**Reply by AnomanderRake69** [reply]
Awesome! My OCP is recognized as a 4p now. njsPC is still not seeing the pump in initial configuration, nor can I add it manually. The IC20 SWCG is also not detected in initial config. Replay attached.

This house was built in 2009, and the OCP label has a date of 4/17/2008, so I guess that's old. But that's why I'm exploring njsPC - while the Pentair functions enough to turn on relays and move valves, the UI is terrible.

And yes, I have a remote control unit and its fully functional. It's labeled as an "EasyTouch", however. Is this relevant?


![handheld](https://user-images.githubusercontent.com/10177469/204809257-2b9dd612-a55b-4f82-9f34-e8f964779adc.jpg)
[replay - 4p.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10123632/replay.-.4p.zip)

**Reply by rstrouse** [reply]
Unfortunately, 2008 is about when they stopped selling this model of EasyTouch so it is old and very much forgotten by most.  njsPC and Nixie will give you way more power.

Yes this remote is relevant.  The only way you can reflect changes through this remote is through the receiver that is connected to the RS485 bus.  It should be a mirror of the OCP firmware.  So if you could capture a replay while changing the speeds for the pump using the remote that would be awesome.

**Reply by AnomanderRake69** [reply]
Thanks for that context, I didn't realize that I was dealing with such an outdated unit. This makes me rethink whether it is a good use of your time to add support for such an outdated unit to njsPC. Given that I am obviously the first one to encounter this issue, I am not keen to see you waste a lot of time on this when I am sure there are other bits of code you'd rather be working on. 

And as I posted above, I'm certainly not going to upgrade my OCP to a newer Pentair unit - the cost is astronomical. I'd rather build myself a Nixie, as it scratches my DIY itch, and is also more cost effective. The wiki page that describes the build process is well done and quite helpful as well.

All that said, I'll leave it to you. I did several pump speed changes with the remote, and have attached the replay logs.
[replay - pump changes with remote.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10126071/replay.-.pump.changes.with.remote.zip)

**Reply by rstrouse** [reply]
I'll have a look at it anyway since I need to solve this mystery for my own edification.   I agree with your assessment and Nixie will give you a much better system than any of the current Pentair systems.  I think you will find that the control, scheduling, and management of your equipment will go beyond what EasyTouch offers and even the new IntelliCenter.  I run an IntelliCenter but part of the reason I got involved in this project was because I upgraded my problematic IntelliTouch to IntelliCenter and it drove me nuts that it was so closed and too many limits.

**Reply by rstrouse** [reply]
Well look at that.  It looks like you are trying to change the settings for a spa side remote.  Can you verify if the menus described in this manual exist starting with page 31 `Settings Menu: IntelliFlo)`.  Could it be that the original setup for your panel was done so unconventionally that they tried to use the is4 remote to control the pump.
https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/automation/easytouch/manual/easytouch-control-system-8-and-4-users-guide.pdf

NOTE:  This should be the manual for an EasyTouch 4PSC from 2008.

**Reply by AnomanderRake69** [reply]
I'm getting the feeling that I have some kind of weird bastardized OCP on my pool. 

The manual you linked is not the right one, I know this because the previous owners were kind enough to leave the paper manual behind. I was unable to find the pdf on Pentair's site, but I did find it here: http://images.inyopools.com/cloud/documents/pentair-controls-easytouch-pool-spa.pdf

p33 in the pdf (marked as p25) in the manual I linked accurately depicts the pump settings screens that I actually have on my OCP. These are the same screens that I attached 11 posts above: https://github.com/rstrouse/nodejs-poolController-dashPanel/discussions/66#discussioncomment-4258481

I have 4 circuits for the pump, and I can set a speed for each circuit. That's the totality of my pump configuration options.

**Reply by rstrouse** [reply]
Yeah this is a strange one.  It appears they took one of the is4 slots and mapped it to the pump settings.  The odd part is that they forgot that there is a hardware interface to that device.  I will go back and look at the is4 dumps from the config.  We may need to look at these messages and map them to the pump settings.  I’ll take a look tomorrow.

**Reply by rstrouse** [reply]
Ok so I pushed some changes that should get the pump data from the controller.  The weird part is that your controller never asks for the current RPM and Wattage.  Booh.  Please pull njsPC then run a replay that reloads the configuration and upload it here.  I may try to figure out the firmware version and tell the stupid thing to get the current running status for the pump.

BTW you still cannot make changes to the speeds from dashPanel but once I see the data from the output I can add it.

**Reply by AnomanderRake69** [reply]
Dude, you totally crushed it. (sorry for the delay - long couple of days at work). 

Both pump and SWCG are now detected on startup! Hopefully this has value to njsPC overall and not just me alone.

Full replay attached.
[replay - success.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10142722/replay.-.success.zip)

**Reply by rstrouse** [reply]
Haven't crushed it yet.  If you pull njsPC you should now be able to set the speeds and more importantly it should reflect the RPM and wattage on the dashPanel.  Amazingly, your version of EasyTouch does not even ask the pump for this information.  I added a bit of code to periodically ping it and return it to njsPC.

I am guessing that there are very few of these panels left in the wild which is why we haven't seen one.  If I were a guesser I would suspect that they did this when they released the VS pumps.  IntelliTouch always supported the 4x160 but EasyTouch did not so they created a Rev B that robbed one of the spa side remotes to store the data.  A newer firmware was later released (Called EasyTouch 2) that finally added support for the addressed pumps.

Please pull njsPC and provide me with another replay so I can verify proper responses from the pump and panel.

**Reply by AnomanderRake69** [reply]
As I suspected then, a bastardized OCP, but done by Pentair themselves, not the pool installer. Quite interesting. Well, all the more reason for me to replace this with a DIY Nixie at some point.

Here's a replay with initial config, as well as attempts to change pump speeds, which was not successful. I am getting RPMs and watts now though.
[replay - pump.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/10143846/replay.-.pump.zip)

**Reply by rstrouse** [reply]
Thanks for that.  I found the issue and fixed it so if you pull njsPC it should now be working for all of it.   I mixed up the array when building the RS485 message to the OCP.

I think there was a very short time between when they were shipping REV A of EasyTouch and upgraded the firmware to EasyTouch 2 where they were shipping a REV B model of the EasyTouch.  Keep in mind there are a couple of things that will not do anything on the ET1.  The first is you cannot set the priming time for the pump from the controller.  Sorry to say it didn't work well on ET2 either.  The other thing is that the max speed settings are only applicable to njsPC.  ET1 doesn't do anything with these.

The really good news is that we have your pump actually reporting its status.  Your ET1 is not even participating in this conversation.  Also, as an aside you can quickly adjust the speeds for existing circuits by clicking on the pump in the dashPanel.  This will allow you to adjust these quickly.  Bear in mind ET1 is even slower than ET2 so if you need to change the speed a lot hold down the up or down arrow and it will increment the values without sending it of to the OCP until you let up.
![image](https://user-images.githubusercontent.com/47839015/205396040-8d5ac0c1-8d65-4027-a21e-bb21bb74e80b.png)

Honestly, the EasyTouch and IntelliTouch boards were good for their time but they are really long in the tooth.  I also suspect that many of these boards are probably long dead.  Their RS485 chips and darlington transistor chips go bad.  Given what I have seen of this panel I would be inclined to do a full on Nixie as well.  It has all the functions of the original but adds a whole lot more functionality.

Another tip would be to provide your latitude and longitude on the General tab under personal information.  This will allow you to create sunrise and sunset schedules if you desire.   Since I am a masochist I wrote a heliotrope into njsPC to track the sun.

Finally, if you uncheck Show as Feature for your Pool Light it will tuck it into the lights section.  For circuits that you want to hide (like AUX 4) you can uncheck this as well and it will remove itself from the display.

**Reply by AnomanderRake69** [reply]
Amazing work @rstrouse! And can I say amazing support too? I honestly hope that this debugging session we just went through will help others as well, but I'm guessing from your last post that I may be one of the few remaining owners of this Frankenstein EasyTouch. 

I too am pretty amazed that this OCP has not died yet. It is 14 years old or so, and has sat in the AZ heat for all that time. 

"Your ET1 is not even participating in this conversation." I LOLed at that at the same time I was impressed by your ingenuity. Even more reason to go full Nixie and unleash the full automation power!

**Reply by rstrouse** [reply]
It is interesting that we have not seen this before.  There may have been one years ago with an ET8 but they were trying to use an IntelliFlo VSF which will absolutely not work on an ET1.  In the end they replaced the ET motherboard to upgrade the firmware to an ET2. 
 These days I would have told them to add another RS485 port to the pi and let Nixie handle the pump.  

Just to give you an idea of the capabilities, njsPC will manage multiple RS485 ports.  This will allow users to have up to 64 pumps, 4 Chlorinators, 64 circuits, unlimited features, unlimited schedules, and a crazy number of sensors.  And if that isn't enough, it opens up your system so you can use tools to keep track of the history, provide automated chemistry dosing, and integrate with your home automation.  I take a quick look at this screen to see if my chems are good.   It rained yesterday so I just need to keep my ORP on trend.
![image](https://user-images.githubusercontent.com/47839015/205409893-e95a4899-e290-4834-98a5-fcad883713ad.png)

**Reply by AnomanderRake69** [reply]
Fantastic. I very much look forward to being able to track my pool more consistently, and those charts are amazing. More sensors and more automation is what I am looking for. My automation is very basic right now, as you can tell from my logs. And to think how much that system cost when it was originally installed. Yikes.

I started shopping for Nixie parts today, so the journey continues...

**Reply by rstrouse** [reply]
Let me know if you have any questions.  I have also outlined my homegrown Chem controller here:

It is shocking how well this works to make sure the pool is balanced.
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller

**Reply by AnomanderRake69** [reply]
LOL. I have so much to do to take my prehistoric controller to 2022 levels. Starting with reading some more - I quickly scanned that link then realized that I need to dedicate some time to carefully reading and understanding and planning - when I'm not distracted with other things.

Thanks for all the detailed writeups in addition to the awesome code.
